File: qq-music-server-lib/src/main/java/name/sitorhy/server/controller/FollowController.java
package name.sitorhy.server.controller;

import name.sitorhy.server.model.ServiceResponse;
import name.sitorhy.server.model.Singer;
import name.sitorhy.server.service.FollowService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;
import reactor.core.publisher.Mono;

import java.util.List;

@RestController
@RequestMapping("/follow")
public class FollowController {
    private FollowService followService;

    @Autowired
    public void setFollowService(FollowService followService) {
        this.followService = followService;
    }

    @GetMapping("/singers")
    public Mono<ServiceResponse<List<Singer>>> getFollowSingers(
            @RequestParam(value = "pageNo", required = false, defaultValue = "1") Long pageNo,
            @RequestParam(value = "pageSize", required = false, defaultValue = "12") Long pageSize
    ) {
        return followService.getFollowSingers(pageNo, pageSize)
                .map((result) -> new ServiceResponse<>(result, true))
                .onErrorResume(ex -> Mono.just(new ServiceResponse<>(ex.getMessage(), false)));
    }
}

File: app-dev-server/src/follow/follow.controller.ts
import { Body, Controller, Get, Post } from '@nestjs/common';
import { FollowService } from './follow.service';

@Controller('follow')
export class FollowController {
  constructor(private readonly followService: FollowService) {}

  @Post('/singers')
  getFollowSingers(@Body() params: { pageNo: number, pageSize: number; }) {
    return this.followService.getFollowSingers(params?.pageNo, params?.pageSize);
  }
}
